### PR TITLE
Introduce GitHub Actions Workflow for Llama Stack Tests

### PR DESCRIPTION
**Pull request by ConnorHack**
# What does this PR do?

Initial implementation of GitHub Actions workflow for automated testing of Llama Stack.

## Key Features
- Automatically runs tests on pull requests and manual dispatch
- Provides support for GPU required model tests
- Reports test results and uploads summaries

### CODE REVIEW

**Comment by ashwinb** [comment]
A couple questions:

- can you attach some results of having run this workflow as you have been experimenting with it? 
- if I needed to run this workflow on an already committed commit on main/ let's say, is there a way to do it? 
- is there a way to manually start running this workflow on PRs?

**Comment by ashwinb** [comment]
(Some of these questions are due to my n00b-ness with everything GitHub :))

**Comment by ConnorHack** [comment]
> A couple questions:
> 
> * can you attach some results of having run this workflow as you have been experimenting with it?

1. [Here is the workflow](https://github.com/ConnorHack/llama-stack/actions/runs/12149365487) being manually run off of [Commit 435f34b](https://github.com/meta-llama/llama-stack/commit/435f34b05e84f1747b28570234f25878cf0b31c4) that was pulled 11 hours ago.  **Note that NO tests were collected for the marks**: `meta_reference` and `llama_3b`:

image

> * if I needed to run this workflow on an already committed commit on main/ let's say, is there a way to do it?
2. Yes there is a way to do it.  You can specify it with the input that is currently labeled `Branch to checkout`.  Just pass in the SHA and it will use it as the checkout reference.  `Branch to checkout` can be used to specify the branch, tag, or SHA.  Here is a picture of the workflow when this occurs:

image

And these are the current inputs (and defaults) supported when manually running a workflow:

image

> * is there a way to manually start running this workflow on PRs?
3. The workflow will automatically run on PRs that change the following files:
```
    paths:
      - 'llama_stack/**/*.py'
      - 'tests/**/*.py'
```

Although answering your question: yes there is a way to also manually run the workflow referencing a PR; as mentioned above just specify the SHA of the top commit of the PR and the workflow will use that when checking out the repository.

File: .github/workflows/gha_workflow_llama_stack_tests.yml [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+          if [ "${MODEL_ID}" = "llama_3b" ] && [ -d "${MODEL_CHECKPOINT_DIR}/${LLAMA_3B_OVERRIDE}" ]; then
+            echo "MODEL_CHECKPOINT_DIR=${MODEL_CHECKPOINT_DIR}/${LLAMA_3B_OVERRIDE}" >> "$GITHUB_ENV"
+          elif [ "${MODEL_ID}" = "llama_8b" ] && [ -d "${MODEL_CHECKPOINT_DIR}/${LLAMA_8B_OVERRIDE}" ]; then
+            echo "MODEL_CHECKPOINT_DIR=${MODEL_CHECKPOINT_DIR}/${LLAMA_8B_OVERRIDE}" >> "$GITHUB_ENV"
+          else
+            echo "MODEL_ID & LLAMA_*B_OVERRIDE are not a valid pairing. Terminating workflow."
+            exit 1
+          fi
+
+      - name: "[DEBUG] Checkpoint update check"
```
Comment (ashwinb) [inline .github/workflows/gha_workflow_llama_stack_tests.yml:191]: can we just do this unconditionally? just echoing something is harmless

File: .github/workflows/gha_workflow_llama_stack_tests.yml [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+        id: install_apt
+        run: |
+          echo "[STEP] Installing 'apt' required packages"
+          sudo apt update -y
+          sudo apt install -y python3 python3-pip npm wget
+
+      - name: "Installing packages with 'curl'"
+        id: install_curl
+        run: |
+          curl -fsSL https://ollama.com/install.sh | sh
```
Comment (ashwinb) [inline .github/workflows/gha_workflow_llama_stack_tests.yml:210]: oh wow, this is a pretty heavy dependency.  let's avoid it for now. I think we can update to use a better container base image without needing any of these installs later.

**Review by ashwinb — APPROVED** [review]
Thank you so much, let's get this in. 

One request: let's not make this automatically run yet. We would like to manually test drive it over the next few days first.

**Comment by ConnorHack** [comment]
@ashwinb Just updated this PR to temporarily not run on PRs until manual verification has been completed.

Testing has been verified to be the same via my fork's [Run #135](https://github.com/ConnorHack/llama-stack/actions/runs/12163681190).

File: .github/workflows/gha_workflow_llama_stack_tests.yml (hunk 1 above)
Comment (ConnorHack) [reply .github/workflows/gha_workflow_llama_stack_tests.yml:193]: I was considering taking this out, but if you find value I can remove the conditional so this always runs for manual dispatch workflows & PRs.

File: .github/workflows/gha_workflow_llama_stack_tests.yml (hunk 2 above)
Comment (ConnorHack) [reply .github/workflows/gha_workflow_llama_stack_tests.yml:212]: We can experiment removing this once this PR lands.  I needed it early on prior to the rework you did, and there's a chance a few of these other dependencies are no longer needed.

**Pull request merged**